### DISCUSSION TITLE
range picker with two months shown on screen

### DISCUSSION THREAD

**Discussion by Makho96** [Hacker0x01/react-datepicker#3839] [Q&A]
Hello guys. I have a question. When there are two months shown when "selectsRange" porp is true, if I change month in one of the calendars with arrow, second month is also changed automatically. Is there any way to separate this two methods?

**Comment by aziyatali** [comment]
@Makho96 Yes, you can do that. All you need to do is creating two independent states.

```
 const [startDate, setStartDate] = useState(new Date());
  const [endDate, setEndDate] = useState(new Date());

  return (
    <div>
      <h1>Select Range</h1>
      <DatePicker
        selectsRange={true}
        startDate={startDate}
        endDate={endDate}
        onChange={(update) => {
          setStartDate(update.startDate);
          setEndDate(update.endDate);
        }}
      />
    </div>
```

Please, if it answers your question, mark as an answer.
